lib/matplotlib/patches.py

```diff
@@ -479,11 +480,15 @@ def get_hatch(self):
         'Return the current hatching pattern'
         return self._hatch
 
-    @artist.allow_rasterization
-    def draw(self, renderer):
-        'Draw the :class:`Patch` to the given *renderer*.'
-        if not self.get_visible():
-            return
+    @contextlib.contextmanager
+    def _prepare_path_drawer(self, renderer):
```

Comment:
An alternative name could be `_bind_draw_path_function`. This would result in

~~~
with self._bind_draw_path_function(renderer) as draw_path:
    ...
~~~

A bit longish but maybe also a bit more telling. I don't have strong feelings on this. Leave it or use it.

Reply:
done